### ISSUE TITLE
Getting a history of previously executed commands

### ISSUE THREAD

**Issue by sean-hill** [webdriverio/webdriverio#618]
Hey there,

I was wondering if there was functionality for getting a history of commands that had executed previously. 

For instance:

	client
		.init()
		.url('http://www.google.com')
		.click('#username')
		.addValue('#password', 'text')
		.pause(2000)
		.getHistory(function(err, history){

			// History would be an array such as
			//
			// [ 'init', 'went to google.com', 'clicked #username', 'typed text into #password', 'paused 2000ms' ]

		})
		.end()

It wouldn't have to be as human readable as that, but you get the idea.
That'd be awesome :)

**Comment by christian-bromann** [comment]
@sean-hill interesting idea. Yeah that would be no problem at all but what do you want to do with these information?

**Comment by sean-hill** [comment]
Hey @christian-bromann, so sometimes with webdriverio, I might encounter an unknown error. I catch these with `client.on('error')`, and then email myself that a problem happened. It'd be nice to know how far the `client` got in its runtime before it encountered an unknown error. Thanks!

**Comment by christian-bromann** [comment]
So it seems that you are using WebdriverIO not for integration testing but more for some automation tasks right?

**Comment by sean-hill** [comment]
That is correct!

**Comment by christian-bromann** [comment]
I see, how about the following structure:

```js
client
    .init()
    .url('http://www.google.com')
    .click('#username')
    .addValue('#password', 'text')
    .pause(2000)
    .getHistory().then(function(history){        
        console.log(history);
    })
    .end();
```

that would output:

```
[{
    command: 'init'
},{
    command: 'click',
    args: ['#username']
},{
    command: 'addValue',
    args: ['#password', 'text']
},{
    command: 'pause',
    args: [2000]
}]
```

I can imagine scenarios where that could be useful for automation and since it is not a big thing to add that, let's do this.

**Comment by sean-hill** [comment]
Woot woot :+1: That looks perfect.

**Comment by christian-bromann** [comment]
I called the command `getCommandHistory` will get shipped with `v3.2.0`

**Comment by christian-bromann** [comment]
Published with `v3.2.0`
